### PR TITLE
fix(header): fix testing utilities and tests

### PR DESCRIPTION
**Pull request by Wondertan**
#2215 was approved and merged too quickly, making some tests fail. This PR fixes that. Expect further a new PR to clean up Validate func and increase coverage for it.

### CODE REVIEW

**Review by renaynay — APPROVED** [review]

File: header/header.go [hunk 1]
```diff
@@ -132,6 +132,12 @@ func (eh *ExtendedHeader) Validate() error {
 		)
 	}
 
+	// ensure data root from raw header matches computed root
```
Comment (renaynay) [inline header/header.go:135]: why the change in order

**Review by renaynay — APPROVED** [review]

File: header/header.go (hunk 1 above)
Comment (Wondertan) [reply header/header.go:135]: I will reorder those things in the next PR anyway. The order of verification matters here as well and ours is slightly off

**Review by vgonkivs — APPROVED** [review]
